From 181622f8879921a7b40411290bd92cc6b5f8efb1 Mon Sep 17 00:00:00 2001
Message-Id: <181622f8879921a7b40411290bd92cc6b5f8efb1.1434670190.git.chang-joon.lee@intel.com>
In-Reply-To: <5c49cb8f884ecf227e44159a5a4931d2c0814d5c.1434670190.git.chang-joon.lee@intel.com>
References: <5c49cb8f884ecf227e44159a5a4931d2c0814d5c.1434670190.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Mon, 1 Jun 2015 13:43:05 +0530
Subject: [PATCH 2/6] Revert "FOR_UPSTREAM [VPG]: drm/i915: Setting the device
 ready during the ULPS exit and enter"

This reverts commit 385aba99a44fe109bf6b68e5258c45314ee9c6ad.
Toggle of DEVICE_READY bit during ULPS ENTRY or EXIT
should not be done. This can cause incorrect behavior
of DSI controller. This patch is being reverted to make sure
DEVICE_READY bit is not toggled during ULPS ENTRY/EXIT sequence

Issue: GMINL-10295
Change-Id: If0a26f31db5036438a60472df560cbe54d643d3e
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index d9eaaac..f5c0ac8 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -213,8 +213,7 @@ static void intel_dsi_device_ready(struct intel_encoder *encoder)
 		val |= pipe ? DELAY_180_PHASE_SHIFT_MIPIC :
 				DELAY_180_PHASE_SHIFT_MIPIA;
 
-	intel_dsi_write_dev_rdy_on_A_and_C(encoder,
-			DEVICE_READY | ULPS_STATE_ENTER);
+	intel_dsi_write_dev_rdy_on_A_and_C(encoder, ULPS_STATE_ENTER);
 
 	/* wait for LP state to go 00 */
 	usleep_range(2500, 3000);
@@ -226,8 +225,7 @@ static void intel_dsi_device_ready(struct intel_encoder *encoder)
 
 	I915_WRITE(MIPI_PORT_CTRL(0), val | LP_OUTPUT_HOLD);
 	usleep_range(1000, 1500);
-	intel_dsi_write_dev_rdy_on_A_and_C(encoder,
-			DEVICE_READY | ULPS_STATE_EXIT);
+	intel_dsi_write_dev_rdy_on_A_and_C(encoder, ULPS_STATE_EXIT);
 
 	/* wait for LP state to goto 11 */
 	usleep_range(2500, 3000);
-- 
1.9.1

